How to find the data where one field is nil?

```
type Something struct {
	Id        bson.ObjectId `storm:"id"`
	CreatedAt time.Time     `storm:"index"`
	DeletedAt *time.Time    `storm:"index"`
}

func TestStorm(t *testing.T) {
	//	db, err := storm.Open("gob.db", storm.Codec(gob.Codec))
	db, err := storm.Open("my.db")
	if err != nil {
		t.Error(err)
	}
	defer db.Close()
	data := Something{
		Id:        bson.NewObjectId(),
		CreatedAt: time.Now(),
	}
	err = db.Save(&data)
	if err != nil {
		t.Error(err)
	}
	t.Logf("data.DeletedAt == nil: %v", data.DeletedAt == nil)
	var list []Something
	err = db.Find("DeletedAt", nil, &list)
	if err != nil {
		t.Error(err)
	}
}
```
result:
=== RUN   TestStorm
--- FAIL: TestStorm (0.02s)
	stom_test.go:33: data.DeletedAt == nil: true
	stom_test.go:37: not found

--- a/finder.go
+++ b/finder.go
@@ -119,10 +119,6 @@ func (n *node) one(tx *bolt.Tx, bucketName, fieldName string, cfg *structConfig,
 
 // Find returns one or more records by the specified index
 func (n *node) Find(fieldName string, value interface{}, to interface{}, options ...func(q *index.Options)) error {
-	if value == nil {
-		return ErrNotFound
-	}
-
 	sink, err := newListSink(n, to)
 	if err != nil {
 		return err
@@ -144,10 +140,10 @@ func (n *node) Find(fieldName string, value interface{}, to interface{}, options
 	}
 
 	field, ok := cfg.Fields[fieldName]
-	if !ok || (!field.IsID && field.Index == "") {
+	if !ok || (!field.IsID && (field.Index == "" || value == nil)) {
 		sink.limit = opts.Limit
 		sink.skip = opts.Skip
-		query := newQuery(n, q.StrictEq(fieldName, value))
+		query := newQuery(n, q.Eq(fieldName, value))
 
 		if opts.Reverse {
 			query.Reverse()

--- a/finder_test.go
+++ b/finder_test.go
@@ -112,6 +112,46 @@ func TestFind(t *testing.T) {
 	assert.NoError(t, err)
 }
 
+func TestFindNil(t *testing.T) {
+	db, cleanup := createDB(t, AutoIncrement())
+	defer cleanup()
+
+	type User struct {
+		ID        int        `storm:"increment"`
+		CreatedAt *time.Time `storm:"index"`
+		DeletedAt *time.Time `storm:"unique"`
+	}
+
+	t1 := time.Now()
+	for i := 0; i < 10; i++ {
+		now := time.Now()
+		var u User
+
+		if i%2 == 0 {
+			u.CreatedAt = &t1
+			u.DeletedAt = &now
+		}
+
+		err := db.Save(&u)
+		assert.NoError(t, err)
+	}
+
+	var users []User
+	err := db.Find("CreatedAt", nil, &users)
+	require.NoError(t, err)
+	require.Len(t, users, 5)
+
+	users = nil
+	err = db.Find("CreatedAt", t1, &users)
+	require.NoError(t, err)
+	require.Len(t, users, 5)
+
+	users = nil
+	err = db.Find("DeletedAt", nil, &users)
+	require.NoError(t, err)
+	require.Len(t, users, 5)
+}
+
 func TestFindIntIndex(t *testing.T) {
 	db, cleanup := createDB(t, AutoIncrement())
 	defer cleanup()

--- a/q/compare.go
+++ b/q/compare.go
@@ -14,6 +14,11 @@ func compare(a, b interface{}, tok token.Token) bool {
 	ak := vala.Kind()
 	bk := valb.Kind()
 	switch {
+	// comparing nil values
+	case (ak == reflect.Ptr || ak == reflect.Slice || ak == reflect.Interface || ak == reflect.Invalid) &&
+		(bk == reflect.Ptr || ak == reflect.Slice || bk == reflect.Interface || bk == reflect.Invalid) &&
+		(!vala.IsValid() || vala.IsNil()) && (!valb.IsValid() || valb.IsNil()):
+		return true
 	case ak >= reflect.Int && ak <= reflect.Int64:
 		if bk >= reflect.Int && bk <= reflect.Int64 {
 			return constant.Compare(constant.MakeInt64(vala.Int()), tok, constant.MakeInt64(valb.Int()))

